src/couch_replicator_manager.erl

```diff
@@ -124,13 +126,25 @@ replication_error(#rep{id = {BaseId, _} = RepId}, Error) ->
     nil ->
         ok;
     #rep_state{rep = #rep{db_name = DbName, doc_id = DocId}} ->
+        add_error_jitter(),
         update_rep_doc(DbName, DocId, [
             {<<"_replication_state">>, <<"error">>},
             {<<"_replication_state_reason">>, to_binary(error_reason(Error))},
             {<<"_replication_id">>, ?l2b(BaseId)}]),
         ok = gen_server:call(?MODULE, {rep_error, RepId, Error}, infinity)
     end.
 
+% Add random delay proportional to the number of replications
+% on current node, in order to prevent a stampede when a source
+% with multiple replication targets fails
+add_error_jitter() ->
+    RepCount = ets:info(?REP_TO_STATE, size),
+    Range = RepCount * 2 * ?AVG_ERROR_DELAY_MSEC,
```

Comment:
Since config macro is called "average error", then made the random uniform range * 2. For example if need an average of 50,  sample from a uniform interval of 0 to 100.